### DISCUSSION TITLE
Unable to Customize TypeScript Naming Strategy for Entities that are auto generated

### DISCUSSION THREAD

**Discussion by Ian-Eduware** [mikro-orm/mikro-orm#5721] [General]
### Describe the bug

I am currently using Mikro-ORM with a TypeScript project, and I've encountered an issue with customizing the naming strategy for entity fields. The entity generator in Mikro-ORM  specifies field names via the fieldName property that match my DB table, but it automatically matched the TypeScript property names based on the field names in the database schema.

I would like to use camelCase for the TypeScript property names instead of the capitalization which is done in my DB table. Despite exploring the naming strategy options provided by Mikro-ORM, I have not found a way to change the naming pattern for TypeScript property names while keeping the database field names unchanged.

**Entity example where I want my TS types to follow a different naming strategy when auto generated**

```js
import { Entity, PrimaryKey, PrimaryKeyProp, Property } from '@mikro-orm/core';

@Entity({ tableName: 'AdminSetting' })
export class AdminSetting {

  [PrimaryKeyProp]?: 'AdminSettingId';

  @PrimaryKey({ fieldName: 'AdminSettingId' })
  AdminSettingId!: number;

  @Property({ fieldName: 'Setting', length: 100 })
  Setting!: string;

  @Property({ fieldName: 'SettingValue', length: 100, nullable: true })
  SettingValue?: string;

  @Property({ fieldName: 'IsEnable' })
  IsEnable!: boolean;

}

```

### Reproduction

1. Create an entity using Mikro-ORM's entity generator with field names specified via fieldName.
2. Observe that TypeScript property names are capitalized based on the field names.
3. Attempt to change the naming strategy to use camelCase for TypeScript property names.

### What driver are you using?

@mikro-orm/mssql

### MikroORM version

^6.2.9

### Node.js version

v22.3.0

### Operating system

Windows 11

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Please provide actual reproduction as code, if you don't like what the entity generator is producing, a reproduction would be an existing schema dump + ORM config you are using.

I believe you want to use this option:

https://mikro-orm.io/api/core/class/AbstractNamingStrategy#columnNameToProperty

**Reply by Ian-Eduware** [reply]
Thank you for pointing me in the right direction. I revisited the `columnNameToProperty` option and it indeed solved the issue I was facing. Initially, I overlooked this option because it wasn't required, and I didn't realize it could be tailored to handle different column naming conventions.

For others who might encounter the same issue, here's the solution that worked for me to standardize database column names into camelCase for TypeScript properties using a custom naming strategy. Particularly useful your database uses many different naming conventions and you want to maintain a consistent camelCase in your TypeScript code.

```typescript
import { AbstractNamingStrategy, NamingStrategy } from '@mikro-orm/core';

export class CustomNameStrategy extends AbstractNamingStrategy implements NamingStrategy {
    classToTableName(entityName: string): string {
        return entityName;  // Keeps the original table name
    }

    joinColumnName(propertyName: string): string {
        return propertyName;  // Keeps the original join column name
    }

    joinKeyColumnName(entityName: string, referencedColumnName?: string): string {
        return `${entityName}_${referencedColumnName || 'id'}`;  // Standard FK naming
    }

    joinTableName(sourceEntity: string, targetEntity: string, propertyName?: string): string {
        return `${sourceEntity}_${targetEntity}`;  // Standard join table naming
    }

    propertyToColumnName(propertyName: string): string {
        return propertyName;  // Keeps the original column name
    }

    referenceColumnName(): string {
        return 'id';  // Standard reference column name
    }

    columnNameToProperty(columnName: string): string {
        return this.toCamelCase(columnName);  // Converts to camelCase
    }

    private toCamelCase(columnName: string): string {
        return columnName
            .replace(/(\bID\b|\bid\b|\bId\b)/g, 'id')  // Normalize 'id' variations to 'id'
            .replace(/_([a-zA-Z])/g, (_, g) => g.toUpperCase())  // Convert underscores to camelCase
            .replace(/([a-z])([A-Z]+)/g, (match, p1, p2) => p1 + p2.charAt(0).toUpperCase() + p2.slice(1).toLowerCase())  // Proper camelCase
            .replace(/([A-Z]+)(?=[A-Z][a-z])/g, (match) => match.charAt(0).toUpperCase() + match.slice(1).toLowerCase())  // Adjust remaining all caps
            .replace(/^[A-Z]/, (match) => match.toLowerCase());  // Ensure first character is lowercase
    }
}
```
